Character.cpp

```diff
@@ -98,20 +95,42 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        if(hitPoints < *initialHitPoints)
+        {
+            hitPoints = *initialHitPoints;
+        }
+        hitPoints += hitPoints * 0.1;
+        initialHitPoints.reset(new int(hitPoints));
+
+        if(armor < *initialArmorLevel)
+        {
+            armor = *initialArmorLevel;
+        }
+        armor += armor * 0.1;
+        initialArmorLevel.reset(new int(armor));
+        
+
+        if(attackDamage < *initialAttackDamage)
+        {    
+            attackDamage = *initialAttackDamage;
+        }
+        attackDamage += attackDamage * 0.1;
+        initialAttackDamage.reset(new int(attackDamage));
```

Comment:
Don't `reset()`
just dereference and assign.
You are doing the same thing with 3 different pairs of variables.
D.R.Y. the pattern.
write a free function and call it 3 times with each pair of variables
- [ ] revise